components/patient-measurements/api/src/main/java/org/phenotips/measurements/internal/AbstractMeasurementHandler.java

```diff
@@ -166,6 +199,51 @@ public void initialize() throws InitializationException
         this.chartConfigurations = this.settingsFactory.loadConfigurationsForMeasurementType(getName());
     }
 
+    @Override
+    public Collection<VocabularyTerm> getAssociatedTerms(Double standardDeviation)
+    {
+        ResourceBundle configuration = ResourceBundle.getBundle("measurementsAssociatedTerms");
+        List<VocabularyTerm> terms = new ArrayList<>();
+
+        if (null == standardDeviation || standardDeviation <= -3) {
+            addResolvedTermsToList(configuration, this.getName(), "extremeBelowNormal", terms);
+        }
+        if (null == standardDeviation || standardDeviation <= -2) {
```

Comment:
After some thinking on this with the team, there are a couple of issues here.

3. I feel like the logic in `MeasurementUtils.getFuzzyValue(double)` can be somehow combined with this.

1. I don't think the way this function is written makes it clear that `addResolvedTermsToList` will be called a certain number of times in certain situations. Initially, the pattern that this was written in caused me to interpret it as a typical "switch-style" function. 

    Something like this might be clearer:
    ```java
Set<String> fuzzys = new HashSet(Arrays.asList(
            "extremeBelowNormal",
            "belowNormal",
            "aboveNormal",
            "extremeAboveNormal"
            ));

if (standardDeviation != null) {
     if (standardDeviation > -3) {
         set.remove("extremeBelowNormal");
     } else if (standarDeviation > -2) {
         set.remove("belowNormal");
     } /* .... */
}

for (String fuzzy : fuzzys) {
     addResolvedTermsToList(/*...*/ fuzzy /*...*/);h ...
}
    ```

2. @matt doesn't like that `addResolvedTermsToList` is very C-like, as in it takes a pointer to a list it's supposed to modify, instead of returning a list that gets appended to the existing list like a proper "Object-Oriented" function.